### ISSUE TITLE
Ionic 3, Android 7: Data notifications not shown + no default sound while app is out of focus

### ISSUE THREAD

**Issue by iranicus** [phonegap/phonegap-plugin-push#2599]
### Expected Behaviour
When a data notification is received on an Android 7 device while the app is open in the background, to wake up the mobile if on standby, vibrate normally (not vibrate for 1ms), and play the default system notification sound (unless device is on vibrate only of course, then vibrate is enough).

### Actual Behaviour
When a data notification is received if the application is in the background or the device is on standby the device vibrates quickly (for like 1ms), no wakeup, no notification is shown, no default system sound played, the notification icon appears in the system tray at the top for a fraction of a second and then disappears. (Works fine on Android 6 devices however, tested with Nexus 5)

#### Steps to Reproduce

1. Install the cordova native plugin:
      a. ionic cordova plugin add phonegap-plugin-push
      b. npm install --save @ionic-native/push

2. Initialise the push plugin (see sample code of initialisation used later)

3. Install the android platform 7.1.0: ionic cordova platform add android@7.1.0

4. Transfer successful build to device, install + open up to initialise the push plugin, and exit the app with pressing the home button to return to the home screen, put device into standby mode.

5. Send a data notification to the Android 7 device (currently using a Samsung S6 Edge+)

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Android 7

#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
Samsung S6 Edge+

#### Cordova CLI version and cordova platform version
**Ionic:**
   ionic (Ionic CLI)  : 4.2.1 (C:\...\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework    : ionic-angular 3.7.1
   @ionic/app-scripts : 3.0.0

**Cordova:**
   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : android 7.1.0, browser 5.0.3, ios 4.5.4
   Cordova Plugins       : no whitelisted plugins (12 plugins total)

**System:**
   Android SDK Tools : 26.1.1 (C:\...\AppData\Local\Android\Sdk)
   NodeJS            : v8.11.1 (C:\nodeJS\node.exe)
   npm               : 5.8.0
   OS                : Windows 10

#### Plugin version
phonegap-plugin-push 2.2.3 "PushPlugin"

#### Sample Push Data Payload
Java backend which uses a third party library called PushRaven by Raudius to simplify building the push notification, currently the notification payload is constructed like so:

```
private Message createMessage(Notification notification) {
        Message raven = new Message();

        // PN payload for all platforms (Android + ios)
        Map<String, String> data = new HashMap<>();
            data.put("title", notification.getMessage().getTitle());
            data.put("body", notification.getMessage().getBody());
            data.put("priority", "high");
            data.put("sound", "default");
            data.put("sender", notification.getFrom()); // custom data
            // data.put("content-available", "1");

        // Android-specific configuration
        AndroidConfig androidConfig = new AndroidConfig().priority(AndroidConfig.Priority.HIGH);

        // iOS-specific payload options (added to iOS payload later)
        Map<String, Object> apsData = new HashMap<>();
        Map<String, Object> alertData = new HashMap<>();
            apsData.put("sound", "default");
            alertData.put("title", notification.getMessage().getTitle());
            alertData.put("body", notification.getMessage().getBody());
            apsData.put("alert", alertData);

        // iOS-specific configuration (setting aps directory in payload)
        Payload iosPayload = new Payload().addAttributeMap("aps", apsData);
        ApnsConfig iosConfig = new ApnsConfig().payload(iosPayload);

        // Message construction
            raven.name("id");
            raven.token(notification.getDeviceId());
            raven.data(data);
            raven.android(androidConfig);
            raven.apns(iosConfig);

        return raven;
    }
```
The plugin follows the standards set out in the google FCM HTTP v1 API with constructing the payload sent to firebase in the following format (as seen on https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages/send):
 ```
{
  "validate_only": boolean,
  "message": {
    object(Message)
  },
} 
```
The final payload which is sent out to FCM is therefore the following (note currently include the payload for iOS devices too, which currently seem to receive the notification fine having tested on iOS9):
```
{
	"validate_only": false,
	"message": {
		"name": "id",
		"apns": {
			"payload": {
				"aps": {
					"alert": {
						"title": "Notification Title",
						"body": "Notification message here !!!"
					},
					"sound": "default"
				}
			}
		},
		"data": {
			"title": "Notification Title",
			"body": "Notification message here !!!",
			"priority": "high",
			"sender": "egY6KzCsSXU:APA91bG7AyOceYAo01-zzO3cXmds73oI-6VnpjsfLFLFrdOighFvd82-NQblWIVg0pPtaSoKRjxL6ndpSv63ZmBOXXDR1U-tE0iAHM245MIXXNuxfOKhObkyyftx62amEIN73Gm3aYxW",
			"sound": "default"
		},
		"android": {
			"priority": "high"
		},
		"token": "eAaL9boJqrk:APA91bFOslfa8DpPdVCy1Y1by0B0eTpN5DRu8ujZmGsy7Ksc6amIlTV0JKVY-b7Jp1Io64tkk4-ocqmrCuQPS82Fm-I6tBPsAEKsYzHZ5IL4_NvdhpQn9k05kCgCJglBGCaXjX4i_2fw"
	}
}
```
In case anyone is wondering about the included objects in the message object it follows the guidelines set out on: https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages

**payload received on mobile device**
Taking a look at the logged out payload when the notification is received on the mobile device:
```
{
	"sound": "default",
	"title": "Notification Title",
	"message": "Notification message here !!!",
	"additionalData": {
		"priority": "high",
		"sender": "eAaL9boJqrk:APA91bFOslfa8DpPdVCy1Y1by0B0eTpN5DRu8ujZmGsy7Ksc6amIlTV0JKVY-b7Jp1Io64tkk4-ocqmrCuQPS82Fm-I6tBPsAEKsYzHZ5IL4_NvdhpQn9k05kCgCJglBGCaXjX4i_2fw",
		"dismissed": false,
		"coldstart": false,
		"foreground": false
	}
}
```

#### Sample Code that illustrates the problem
I would imagine the issue could be to do with the notification payload (but currently works fine with Android 6), or how I am initialising the push plugin in the mobile app, my initialisation code in the app:

  ```
    ...
    platform.ready().then(() => {
      // platform ready, plugins available for use
      this.pushSetup();
    }).catch((err: any) => console.error('platform ready error: ', err));
}

interface NotificationInterface {
     sound: string,
     title?: string,
     message: string,
     additionalData: {
       priority: string,
       sender: string,
       coldstart: boolean,
       foreground: boolean
     }
}

private pushSetup() {
    console.log('Home::pushSetup - platform identified: ', this.platform.platforms());
    if (!this.platform.is('cordova')) {
      // NOTE: when running on a mobile device cordova will be one of the platforms being used,
      // i.e. on android platforms array will be: cordova, mobile, android
      console.warn('Push notification init skipped due to us using FCM instead of Ionic Push Server. ' +
        'Therefore running app in browser is limited, run on physical mobile devices instead.');
      alert('Push notifications not initialised using test deviceId, see console warning.');
      this.deviceID = "test-device-id";
      return;
    }

    // Channel creation only required for Android 7+ (Nougat and Oreo currently)
    if (this.platform.is('android')) {
      this.push.createChannel(
        {
          id: 'myNotificationChannel',
          description: 'push notifications channel description here',
          importance: 5,
          sound: 'default'
        }
      ).then(res => {
        console.log('PushServiceProvider::configuration - Channel created: ', res);
        // Return a list of currently configured channels (should be one due to init() use
        this.push.listChannels().then((channels) => console.log('List of channels: ', channels));
      }).catch(e => {
        console.error('PushServiceProvider::configuration - Channel creation error: ', JSON.stringify(e));
      })
    } else {
      console.warn('Home::pushSetup - Not identified as an android device, skipping channel creation');
    }
    const options: PushOptions = {
      // sound set to false, default system sound will play
      android: {
        vibrate: true,
        sound: false
      },
      ios: {
        alert: true,
        badge: true,
        sound: true
      },
      windows: {},
      browser: {
        pushServiceURL: 'http://push.api.phonegap.com/v1/push'
      }
    };

    // initialises a channel by default with the channel name PushPluginChannel
    const pushObject: PushObject = this.push.init(options);

    // to check if the app has permission to send push notifications
    this.push.hasPermission().then((res: any) => {
      if (res.isEnabled) {
        console.log('Push notification send permission granted');
      } else {
        console.warn('Push notification send permission NOT granted');
      }
    });

    console.log('###Subscribing to errors regarding phonegap-push-plugin');
    this.subscriptions.push(pushObject.on('error').subscribe((error: any) => {
      console.error('Error with push plugin: ', error);
      alert('Error with push plugin'); // added for iOS debugging
    }));

    console.log('###Registering device');
    this.subscriptions.push(pushObject.on('registration').subscribe((registration: RegistrationInterface) => {
      console.log('registration: ', registration);
    }));

    console.log('###Subscribing to receive push notifications');
    this.subscriptions.push(pushObject.on('notification').subscribe((notification: NotificationInterface) => {
        console.log('notification received, full response: ', notification);
      }, (err: any) => {
        console.log('Failed to subscribe to receive push notifications, details: ', err);
        alert('Subscription to receive PNs Failed');
      }
    ));
  }
```

#### Logs taken while reproducing problem
So for the following logs I used: *adb logcat | grep Push*
I have also attached a full adb logcat file of Android 7 (not filtering on Push) when the app is in the background and the device is on standby/sleep (when power button is pressed and screen is off).
[logsAndroid7.txt](https://github.com/phonegap/phonegap-plugin-push/files/2522038/logsAndroid7.txt)

**Android 7 log when notification is received (app in background)**
```
--- LOG START
10-26 12:35:02.780 22120 23881 D Push_FCMService: onMessage - from: 1027180086053
10-26 12:35:02.782 22120 23881 D Push_FCMService: sender id = 1027180086053
10-26 12:35:02.782 22120 23881 D Push_FCMService: normalize extras
10-26 12:35:02.782 22120 23881 D Push_FCMService: key = priority
10-26 12:35:02.783 22120 23881 D Push_FCMService: replace key priority with priority
10-26 12:35:02.783 22120 23881 D Push_FCMService: key = sender
10-26 12:35:02.783 22120 23881 D Push_FCMService: replace key sender with sender
10-26 12:35:02.783 22120 23881 D Push_FCMService: key = body
10-26 12:35:02.783 22120 23881 D Push_FCMService: replace key body with message
10-26 12:35:02.784 22120 23881 D Push_FCMService: no locale found for key = message, error Value Notification of type java.lang.String cannot be converted to JSONObject
10-26 12:35:02.784 22120 23881 D Push_FCMService: key = sound
10-26 12:35:02.784 22120 23881 D Push_FCMService: replace key sound with sound
10-26 12:35:02.784 22120 23881 D Push_FCMService: key = title
10-26 12:35:02.784 22120 23881 D Push_FCMService: replace key title with title
10-26 12:35:02.785 22120 23881 D Push_FCMService: no locale found for key = title, error Value Notification of type java.lang.String cannot be converted to JSONObject
10-26 12:35:02.785 22120 23881 D Push_FCMService: background
10-26 12:35:02.785 22120 23881 D Push_FCMService: message =[Notification message here !!!]
10-26 12:35:02.785 22120 23881 D Push_FCMService: title =[Notification Title]
10-26 12:35:02.785 22120 23881 D Push_FCMService: contentAvailable =[null]
10-26 12:35:02.785 22120 23881 D Push_FCMService: forceStart =[null]
10-26 12:35:02.785 22120 23881 D Push_FCMService: create notification
10-26 12:35:02.787 22120 23881 E Push_FCMService: Number format exception - Error parsing notId: null
10-26 12:35:02.798 22120 23881 D Push_FCMService: stored icon=null
10-26 12:35:02.798 22120 23881 D Push_FCMService: stored iconColor=null
10-26 12:35:02.798 22120 23881 D Push_FCMService: stored sound=false
10-26 12:35:02.798 22120 23881 D Push_FCMService: stored vibrate=true
10-26 12:35:02.798 22120 23881 D Push_FCMService: no icon resource found - using application icon
10-26 12:35:02.804 22120 23881 D Push_FCMService: create actions: with in-line
10-26 12:35:02.943 22120 22120 D Push_DismissedHandler: PushDismissedHandler = Bundle[mParcelledData.dataSize=716]
10-26 12:35:02.943 22120 22120 D Push_DismissedHandler: not id = 0
--- LOG END
```
**Android 7 when the user receives a notification (app open in foreground)**
```
10-26 12:47:12.776 22120 26280 D Push_FCMService: onMessage - from: 1027180086053
--- LOG START
10-26 12:47:12.776 22120 26280 D Push_FCMService: sender id = 1027180086053
10-26 12:47:12.776 22120 26280 D Push_FCMService: normalize extras
10-26 12:47:12.776 22120 26280 D Push_FCMService: key = priority
10-26 12:47:12.776 22120 26280 D Push_FCMService: replace key priority with priority
10-26 12:47:12.776 22120 26280 D Push_FCMService: key = sender
10-26 12:47:12.777 22120 26280 D Push_FCMService: replace key sender with sender
10-26 12:47:12.777 22120 26280 D Push_FCMService: key = body
10-26 12:47:12.777 22120 26280 D Push_FCMService: replace key body with message
10-26 12:47:12.777 22120 26280 D Push_FCMService: no locale found for key = message, error Value yes of type java.lang.String cannot be converted to JSONObject
10-26 12:47:12.777 22120 26280 D Push_FCMService: key = sound
10-26 12:47:12.777 22120 26280 D Push_FCMService: replace key sound with sound
10-26 12:47:12.777 22120 26280 D Push_FCMService: key = title
10-26 12:47:12.777 22120 26280 D Push_FCMService: replace key title with title
10-26 12:47:12.777 22120 26280 D Push_FCMService: no locale found for key = title, error Value Person-Response of type java.lang.String cannot be converted to JSONObject
10-26 12:47:12.777 22120 26280 D Push_FCMService: foreground
10-26 12:47:12.777 22120 26280 D Push_Plugin: convert extras to json
10-26 12:47:12.777 22120 26280 D Push_Plugin: key = priority
10-26 12:47:12.777 22120 26280 D Push_Plugin: key = sender
10-26 12:47:12.777 22120 26280 D Push_Plugin: key = sound
10-26 12:47:12.777 22120 26280 D Push_Plugin: key = title
10-26 12:47:12.777 22120 26280 D Push_Plugin: key = message
10-26 12:47:12.777 22120 26280 D Push_Plugin: key = coldstart
10-26 12:47:12.778 22120 26280 D Push_Plugin: key = foreground
10-26 12:47:12.778 22120 26280 V Push_Plugin: extrasToJSON: {"sound":"default","title":"Response","message":"yes","additionalData":{"priority":"high","sender":"personA,reg1","coldstart":false,"foreground":true}}
--- LOG END
```
**(No issues on this version) Android 6 log when notification is received (app in background)**
```
--- LOG START
10-26 12:52:41.812  6325 17581 D Push_FCMService: onMessage - from: 1027180086053
10-26 12:52:41.822  6325 17581 D Push_FCMService: sender id = 1027180086053
10-26 12:52:41.822  6325 17581 D Push_FCMService: normalize extras
10-26 12:52:41.822  6325 17581 D Push_FCMService: key = priority
10-26 12:52:41.822  6325 17581 D Push_FCMService: replace key priority with priority
10-26 12:52:41.822  6325 17581 D Push_FCMService: key = sender
10-26 12:52:41.822  6325 17581 D Push_FCMService: replace key sender with sender
10-26 12:52:41.822  6325 17581 D Push_FCMService: key = body
10-26 12:52:41.822  6325 17581 D Push_FCMService: replace key body with message
10-26 12:52:41.832  6325 17581 D Push_FCMService: no locale found for key = message, error Value Notification of type java.lang.String cannot be converted to JSONObject
10-26 12:52:41.832  6325 17581 D Push_FCMService: key = sound
10-26 12:52:41.832  6325 17581 D Push_FCMService: replace key sound with sound
10-26 12:52:41.832  6325 17581 D Push_FCMService: key = title
10-26 12:52:41.832  6325 17581 D Push_FCMService: replace key title with title
10-26 12:52:41.832  6325 17581 D Push_FCMService: no locale found for key = title, error Value Notification of type java.lang.String cannot be converted to JSONObject
10-26 12:52:41.842  6325 17581 D Push_FCMService: background
10-26 12:52:41.842  6325 17581 D Push_FCMService: message =[Notification message here !!!]
10-26 12:52:41.842  6325 17581 D Push_FCMService: title =[Notification Title]
10-26 12:52:41.842  6325 17581 D Push_FCMService: contentAvailable =[null]
10-26 12:52:41.842  6325 17581 D Push_FCMService: forceStart =[null]
10-26 12:52:41.842  6325 17581 D Push_FCMService: create notification
10-26 12:52:41.842  6325 17581 E Push_FCMService: Number format exception - Error parsing notId: Invalid int: "null"
10-26 12:52:41.962  6325 17581 D Push_FCMService: stored icon=null
10-26 12:52:41.972  6325 17581 D Push_FCMService: stored iconColor=null
10-26 12:52:41.972  6325 17581 D Push_FCMService: stored sound=false
10-26 12:52:41.972  6325 17581 D Push_FCMService: stored vibrate=true
10-26 12:52:41.972  6325 17581 D Push_FCMService: no icon resource found - using application icon
10-26 12:52:42.022  6325 17581 D Push_FCMService: create actions: with in-line
```
**interacted with the push notification received on Android 6 while app is in the background, opening the app upon interaction and the following logging appears**
```
10-26 12:54:00.892  1572  3040 W ActivityManager: startActivity called from non-Activity context; forcing Intent.FLAG_ACTIVITY_NEW_TASK for: Intent { flg=0x24000000 cmp=com.project/com.adobe.phonegap.push.PushHandlerActivity (has extras) }
10-26 12:54:00.912  1572  3040 V WindowManager: addAppToken: AppWindowToken{bfb6e32 token=Token{2f4063d ActivityRecord{8d1b94 u0 com.project/com.adobe.phonegap.push.PushHandlerActivity t278}}} to stack=1 task=278 at 1
10-26 12:54:01.022  6325  6325 D Push_HandlerActivity: not id = 0
10-26 12:54:01.022  6325  6325 V Push_HandlerActivity: onCreate
10-26 12:54:01.022  6325  6325 D Push_HandlerActivity: callback = null
10-26 12:54:01.022  6325  6325 D Push_HandlerActivity: dismissed = false
10-26 12:54:01.032  6325  6325 D Push_Plugin: convert extras to json
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = priority
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = sender
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = actionCallback
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = sound
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = title
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = dismissed
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = message
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = coldstart
10-26 12:54:01.032  6325  6325 D Push_Plugin: key = foreground
10-26 12:54:01.032  6325  6325 V Push_Plugin: extrasToJSON: {"sound":"default","title":"Notification Title","message":"Notification message here !!!","additionalData":{"priority":"high","sender":"eAaL9boJqrk:APA91bFOslfa8DpPdVCy1Y1by0B0eTpN5DRu8ujZmGsy7Ksc6amIlTV0JKVY-b7Jp1Io64tkk4-ocqmrCuQPS82Fm-I6tBPsAEKsYzHZ5IL4_NvdhpQn9k05kCgCJglBGCaXjX4i_2fw","dismissed":false,"coldstart":false,"foreground":false}}
10-26 12:54:01.042  6325  6325 D Push_HandlerActivity: bringToForeground = true
10-26 12:54:01.062  6325  6325 D Push_HandlerActivity: isPushPluginActive = true
10-26 12:54:01.062  6325  6325 D Push_HandlerActivity: don't want main activity
--- LOG END
```

One thing I did notice was that in the Android 7 logs there was line:
```
10-26 12:35:02.943 22120 22120 D Push_DismissedHandler: PushDismissedHandler = Bundle[mParcelledData.dataSize=716]
10-26 12:35:02.943 22120 22120 D Push_DismissedHandler: not id = 0
```
This did not appear in the logs for Android 6, and might explain the odd behavior on Android 7.

**Further information**
**- package.json**
```
{
  "name": "MyIonicProject",
  "version": "0.0.1",
  "author": "Ionic Framework",
  "homepage": "http://ionicframework.com/",
  "private": true,
  "scripts": {
    "clean": "ionic-app-scripts clean",
    "build": "ionic-app-scripts build",
    "lint": "ionic-app-scripts lint",
    "ionic:build": "ionic-app-scripts build",
    "ionic:serve": "ionic-app-scripts serve",
    "test": "karma start ./test-config/karma.conf.js",
    "test-ci": "karma start ./test-config/karma.conf.js --single-run",
    "test-coverage": "karma start ./test-config/karma.conf.js --coverage",
    "e2e": "npm run e2e-update && npm run e2e-test",
    "e2e-test": "protractor ./test-config/protractor.conf.js",
    "e2e-update": "webdriver-manager update --standalone false --gecko false"
  },
  "dependencies": {
    "@angular/common": "4.4.3",
    "@angular/compiler": "4.4.3",
    "@angular/compiler-cli": "4.4.3",
    "@angular/core": "4.4.3",
    "@angular/forms": "4.4.3",
    "@angular/http": "4.4.3",
    "@angular/platform-browser": "4.4.3",
    "@angular/platform-browser-dynamic": "4.4.3",
    "@ionic-native-mocks/network": "^2.0.7",
    "@ionic-native-mocks/screen-orientation": "^2.0.7",
    "@ionic-native/core": "^4.5.2",
    "@ionic-native/keyboard": "^4.7.0",
    "@ionic-native/network": "^4.7.0",
    "@ionic-native/push": "^4.16.0",
    "@ionic-native/screen-orientation": "^4.7.0",
    "@ionic-native/splash-screen": "^4.5.2",
    "@ionic-native/status-bar": "^4.5.2",
    "@ionic-native/vibration": "^4.7.0",
    "@ionic/storage": "^2.0.1",
    "android-versions": "^1.3.0",
    "cordova-android": "7.1.0",
    "cordova-browser": "^5.0.3",
    "cordova-ios": "^4.5.4",
    "cordova-plugin-compat": "^1.2.0",
    "cordova-plugin-device": "^1.1.7",
    "cordova-plugin-network-information": "^2.0.1",
    "cordova-plugin-screen-orientation": "^3.0.1",
    "cordova-plugin-settings-hook": "^0.2.7",
    "cordova-plugin-splashscreen": "^4.1.0",
    "cordova-plugin-statusbar": "^2.4.1",
    "cordova-plugin-vibration": "^3.0.1",
    "cordova-plugin-whitelist": "^1.3.3",
    "cordova-sqlite-evcore-extbuild-free": "^0.9.2",
    "cordova-support-google-services": "^1.1.0",
    "es6-promise-plugin": "^4.2.2",
    "fs": "0.0.1-security",
    "ionic-angular": "3.7.1",
    "ionic-mocks": "^1.3.0",
    "ionic-plugin-keyboard": "^2.2.1",
    "ionicons": "3.0.0",
    "moment": "^2.20.1",
    "phonegap-plugin-multidex": "^1.0.0",
    "phonegap-plugin-push": "^2.2.3",
    "rxjs": "5.4.3",
    "sw-toolbox": "3.6.0",
    "zone.js": "0.8.18"
  },
  "devDependencies": {
    "@angular/cli": "^1.6.1",
    "@ionic/app-scripts": "3.0.0",
    "@types/jasmine": "^2.5.45",
    "@types/node": "^8.0.54",
    "angular2-template-loader": "^0.6.2",
    "html-loader": "^0.5.1",
    "istanbul-instrumenter-loader": "^3.0.0",
    "jasmine": "^2.8.0",
    "jasmine-spec-reporter": "^4.2.1",
    "karma": "^1.7.1",
    "karma-chrome-launcher": "^2.2.0",
    "karma-coverage-istanbul-reporter": "^1.3.0",
    "karma-jasmine": "^1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-sourcemap-loader": "^0.3.7",
    "karma-webpack": "^2.0.6",
    "null-loader": "^0.1.1",
    "protractor": "^5.2.0",
    "ts-loader": "^3.2.0",
    "ts-node": "^3.3.0",
    "typescript": "2.3.4"
  },
  "description": "An Ionic project",
  "cordova": {
    "plugins": {
      "cordova-plugin-device": {},
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-whitelist": {},
      "ionic-plugin-keyboard": {},
      "cordova-sqlite-evcore-extbuild-free": {},
      "cordova-plugin-vibration": {},
      "cordova-plugin-screen-orientation": {},
      "cordova-plugin-network-information": {},
      "phonegap-plugin-push": {
        "ANDROID_SUPPORT_V13_VERSION": "27.+",
        "FCM_VERSION": "11.6.2"
      }
    },
    "platforms": [
      "browser",
      "ios",
      "android"
    ]
  }
}
```

**- config.xml**
```
<?xml version='1.0' encoding='utf-8'?>
<widget id="com.project" version="0.0.1" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
    <name>Ionic Project</name>
    <description>An new ionic project with phone gap push</description>
    <author email="hi@ionicframework" href="http://ionicframework.com/">Ionic Framework Team</author>
    <content src="index.html" />
    <access origin="*" />
    <allow-navigation href="http://ionic.local/*" />
    <allow-intent href="http://*/*" />
    <allow-intent href="https://*/*" />
    <allow-intent href="tel:*" />
    <allow-intent href="sms:*" />
    <allow-intent href="mailto:*" />
    <allow-intent href="geo:*" />
    <preference name="webviewbounce" value="false" />
    <preference name="UIWebViewBounce" value="false" />
    <preference name="DisallowOverscroll" value="true" />
    <preference name="android-minSdkVersion" value="19" />
    <preference name="BackupWebStorage" value="none" />
    <preference name="SplashMaintainAspectRatio" value="true" />
    <preference name="FadeSplashScreenDuration" value="300" />
    <preference name="SplashShowOnlyFirstTime" value="false" />
    <preference name="phonegap-version" value="cli-7.1.0" />
    <platform name="android">
        <allow-intent href="market:*" />
        <resource-file src="google-services.json" target="app/google-services.json" />
        <icon density="mdpi" src="resources/android/icon/mdpi.png" />
        <icon density="hdpi" src="resources/android/icon/hdpi.png" />
        <icon density="xhdpi" src="resources/android/icon/xhdpi.png" />
        <icon density="xxhdpi" src="resources/android/icon/xxhdpi.png" />
        <icon density="xxxhdpi" src="resources/android/icon/xxxhdpi.png" />
        <splash density="port-ldpi" src="resources/android/splash/splash-port-ldpi.png" />
        <splash density="port-mdpi" src="resources/android/splash/splash-port-mdpi.png" />
        <splash density="port-hdpi" src="resources/android/splash/splash-port-hdpi.png" />
        <splash density="port-xhdpi" src="resources/android/splash/splash-port-xhdpi.png" />
        <splash density="port-xxhdpi" src="resources/android/splash/splash-port-xxhdpi.png" />
        <splash density="port-xxxhdpi" src="resources/android/splash/splash-port-xxxhdpi.png" />
        <preference name="SplashScreenDelay" value="10000" />
        <preference name="SplashScreen" value="screen" />
        <icon density="ldpi" src="resources/android/icon/drawable-ldpi-icon.png" />
        <icon density="mdpi" src="resources/android/icon/drawable-mdpi-icon.png" />
        <icon density="hdpi" src="resources/android/icon/drawable-hdpi-icon.png" />
        <icon density="xhdpi" src="resources/android/icon/drawable-xhdpi-icon.png" />
        <icon density="xxhdpi" src="resources/android/icon/drawable-xxhdpi-icon.png" />
        <icon density="xxxhdpi" src="resources/android/icon/drawable-xxxhdpi-icon.png" />
        <splash density="land-ldpi" src="resources/android/splash/drawable-land-ldpi-screen.png" />
        <splash density="land-mdpi" src="resources/android/splash/drawable-land-mdpi-screen.png" />
        <splash density="land-hdpi" src="resources/android/splash/drawable-land-hdpi-screen.png" />
        <splash density="land-xhdpi" src="resources/android/splash/drawable-land-xhdpi-screen.png" />
        <splash density="land-xxhdpi" src="resources/android/splash/drawable-land-xxhdpi-screen.png" />
        <splash density="land-xxxhdpi" src="resources/android/splash/drawable-land-xxxhdpi-screen.png" />
        <splash density="port-ldpi" src="resources/android/splash/drawable-port-ldpi-screen.png" />
        <splash density="port-mdpi" src="resources/android/splash/drawable-port-mdpi-screen.png" />
        <splash density="port-hdpi" src="resources/android/splash/drawable-port-hdpi-screen.png" />
        <splash density="port-xhdpi" src="resources/android/splash/drawable-port-xhdpi-screen.png" />
        <splash density="port-xxhdpi" src="resources/android/splash/drawable-port-xxhdpi-screen.png" />
        <splash density="port-xxxhdpi" src="resources/android/splash/drawable-port-xxxhdpi-screen.png" />
        <hook src="hooks/maven_swap.sh" type="before_build" />
    </platform>
    <platform name="ios">
        <allow-intent href="itms:*" />
        <allow-intent href="itms-apps:*" />
        <resource-file src="GoogleService-Info.plist" />
        <icon height="57" src="resources/ios/icon/57.png" width="57" />
        <icon height="114" src="resources/ios/icon/114.png" width="114" />
        <icon height="40" src="resources/ios/icon/40.png" width="40" />
        <icon height="80" src="resources/ios/icon/80.png" width="80" />
        <icon height="120" src="resources/ios/icon/120.png" width="120" />
        <icon height="50" src="resources/ios/icon/50.png" width="50" />
        <icon height="100" src="resources/ios/icon/100.png" width="100" />
        <icon height="60" src="resources/ios/icon/60.png" width="60" />
        <icon height="180" src="resources/ios/icon/180.png" width="180" />
        <icon height="72" src="resources/ios/icon/72.png" width="72" />
        <icon height="144" src="resources/ios/icon/144.png" width="144" />
        <icon height="76" src="resources/ios/icon/76.png" width="76" />
        <icon height="152" src="resources/ios/icon/152.png" width="152" />
        <icon height="167" src="resources/ios/icon/167.png" width="167" />
        <icon height="29" src="resources/ios/icon/29.png" width="29" />
        <icon height="58" src="resources/ios/icon/58.png" width="58" />
        <icon height="87" src="resources/ios/icon/87.png" width="87" />
        <splash height="1136" src="resources/ios/splash/Default-568h@2x~iphone.png" width="640" />
        <splash height="1334" src="resources/ios/splash/Default-667h.png" width="750" />
        <splash height="2208" src="resources/ios/splash/Default-736h.png" width="1242" />
        <splash height="1242" src="resources/ios/splash/Default-Landscape-736h.png" width="2208" />
        <splash height="1536" src="resources/ios/splash/Default-Landscape@2x~ipad.png" width="2048" />
        <splash height="2048" src="resources/ios/splash/Default-Landscape@~ipadpro.png" width="2732" />
        <splash height="768" src="resources/ios/splash/Default-Landscape~ipad.png" width="1024" />
        <splash height="2048" src="resources/ios/splash/Default-Portrait@2x~ipad.png" width="1536" />
        <splash height="2732" src="resources/ios/splash/Default-Portrait@~ipadpro.png" width="2048" />
        <splash height="1024" src="resources/ios/splash/Default-Portrait~ipad.png" width="768" />
        <splash height="960" src="resources/ios/splash/Default@2x~iphone.png" width="640" />
        <splash height="480" src="resources/ios/splash/Default~iphone.png" width="320" />
        <splash height="2732" src="resources/ios/splash/Default@2x~universal~anyany.png" width="2732" />
    </platform>
    <platform name="browser">
        <preference name="SplashScreen" value="screen" />
        <preference name="ShowSplashScreen" value="false" />
        <preference name="SplashScreenBackgroundColor" value="#000" />
        <preference name="SplashScreenDelay" value="3000" />
    </platform>
    <plugin name="cordova-plugin-device" spec="^1.1.7" />
    <plugin name="cordova-plugin-splashscreen" spec="^4.1.0" />
    <plugin name="cordova-plugin-statusbar" spec="^2.4.1" />
    <plugin name="cordova-plugin-whitelist" spec="^1.3.3" />
    <plugin name="cordova-sqlite-evcore-extbuild-free" spec="^0.9.2" />
    <plugin name="cordova-plugin-vibration" spec="^3.0.1" />
    <plugin name="cordova-plugin-screen-orientation" spec="^3.0.1" />
    <plugin name="ionic-plugin-keyboard" spec="^2.2.1" />
    <plugin name="cordova-plugin-network-information" spec="^2.0.1" />
    <plugin name="phonegap-plugin-push" spec="^2.2.3">
        <variable name="ANDROID_SUPPORT_V13_VERSION" value="27.+" />
        <variable name="FCM_VERSION" value="11.6.2" />
    </plugin>
    <preference name="SplashScreen" value="screen" />
    <preference name="SplashScreenDelay" value="3000" />
    <engine name="browser" spec="^5.0.3" />
    <engine name="ios" spec="^4.5.4" />
    <engine name="android" spec="7.1.0" />
</widget>
```

If anyone could provide some insight into the issue it would be appreciated, I've been stuck on this for a couple of weeks now having generated clean ionic projects via the ionic cli for ionic versions 3 and 4 which both ended up with the same Android 7 behavior.
Cheers.

**Comment by iranicus** [comment]
I noticed that when I manually clear the notification away on Android 6, this is when the PushDismissedHandler is called, so it seems like for some reason on Android 7 there is something which is causing this to be called as soon as the notification is received.

**Comment by iranicus** [comment]
I will close this off since the current behavior of having either just sound or vibration is fine with us (and we now receive notifications properly), I noticed on iOS that we indeed get both but for Android we can only get one or the other in which case we have gone with the vibration approach, thus we don't have the notification object included in the payload and only the data and android objects. Currently if the notification object is included (the one shown prior) then we only get the sound and no vibration, at least we are able to choose from them. Thanks now.
